### ISSUE TITLE
Gekko stops working after few hours

### ISSUE THREAD

**Issue by shameed123** [askmike/gekko#1316]
I have been able to setup gekko, backtest strategies, create my own strategies and indicators, and everything seems to be working.

I have also been able to setup live gekkos for paper trading live with my strategies.  They seem to be working fine for as long as I check (30 min after setting up).  Then I leave the system alone.  Next morning, I  am not able to connect to gekko.

The weird part is that when I run "ps ax", it shows node running and some child processes I believe for the live gekkos.  

```
11982 pts/1    S+     0:00 grep --color=auto node
16437 ?        Sl     0:42 node gekko.js --ui
16537 ?        Sl     1:39 /usr/bin/nodejs /root/gekko/core/workers/pipeline/child
16547 ?        Sl     0:13 /usr/bin/nodejs /root/gekko/core/workers/pipeline/child
```
Then I try to connect from my browser and it doesn't.  I run "ps ax" again and it does not show the parent node process anymore, but it shows the child.

```
12034 pts/1    S+     0:00 grep --color=auto node
16547 ?        Sl     0:13 /usr/bin/nodejs /root/gekko/core/workers/pipeline/child
```
Basically I haven't had success with running gekko for long term.

If I restart gekko, the live gekkos are gone, they don't persist for whatever reason.  However I still  have history data that I had gathered before for backtesting.

Has anyone had this kind of issue.  Any help will be appreciated.

**Comment by askmike** [comment]
Could you tell more about how you run gekko (where do execute the `node gekko --ui` command) and what environment you are running on (server, laptop, raspberry pi, etc)?

If you for example SSH into a server and start `node gekko --ui` straight from the terminal the process is bound to the SSH session, meaning as soon as SSH drops the Gekko server will stop.

---

> The weird part is that when I run "ps ax", it shows node running and some child processes I believe for the live gekkos.

When you run `node gekko --ui` a single "server controller" process is started, if you start live gekkos they are executed through child processes. Similar to the sandbox model used by your browser (if you open a new tab in firefox or chrome, everything is run in it's own subprocess under the hood).

**Comment by shameed123** [comment]
Thanks for the response.  I run it on a server.  I had tried disowning the process before shutting down the ssh connection and I even tried to run it as daemon using daemon library from node.  But nothing helped.

However nohup seems to work.  It successfully ran overnight.

Thanks.

**Comment by askmike** [comment]
Great, other options would be to use the linux tool screen or the nodejs tool pm2: https://github.com/Unitech/pm2
